terraform/modules/psn/main.tf

```diff
@@ -13,22 +13,49 @@ variable "subnet_ids" {
   description = "The ID of one or more subnets in which to create a network interface for the endpoint."
 }
 
-variable "security_group_ids" {
+variable "security_group_id" {
   type        = "list"
-  description = "The ID of one or more security groups to associate with the network interface."
+  description = "The security group to allow access to the PSN VPC Endpoint."
 }
 
 resource "aws_vpc_endpoint" "psn_service" {
   vpc_id            = "${var.vpc_id}"
   service_name      = "${var.vpc_endpoint}"
   vpc_endpoint_type = "Interface"
 
-  security_group_ids = ["${var.security_group_ids}"]
+  security_group_ids = ["${aws.security_group.psn_endpoint.id}"]
 
   subnet_ids          = ["${var.subnet_ids}"]
   private_dns_enabled = false
 }
 
+resource "aws_security_group" "psn_endpoint" {
+  name        = "psn-endpoint"
+  description = "The PSN VPC Endpoint"
+}
+
+resource "aws_security_group_rule" "psn_ingress_from_worker" {
+  security_group_id = "${aws_security_group.psn_endpoint.id}"
+
+  type      = "ingress"
+  protocol  = "tcp"
+  from_port = 3128
+  to_port   = 3128
+
+  source_security_group_id = "${var.security_group_id}"
+}
+
+resource "aws_security_group_rule" "worker_egress_to_psn" {
```

Comment:
so this rule allows stuff on the PSN to make connections into our services with destination port 3128?

Reply:
as far as i understand, it's an `egress` rule attached to `"${var.security_group_id}"` (ie the worker security group).  I think terraform uses `source_security_group_id` whether it's actually a source or destination 😕.  (CloudFormation is better in this respect.)